conda/lock.py

```diff
@@ -61,11 +73,44 @@ def __enter__(self):
                 return self
 
         stdoutlog.error("Exceeded max retries, giving up")
-        raise LockError(lockstr % self.lock_path)
+        raise LockError(LOCKSTR % self.lock_path)
 
     def __exit__(self, exc_type, exc_value, traceback):
         try:
             os.rmdir(self.lock_path)
             os.rmdir(self.path)
         except OSError:
             pass
+
+
+class FileLock(object):
+    """
+    Context manager to handle locks.
+    """
+    def __init__(self, filepath, retries=10):
+        self.filepath = abspath(filepath)
+        self.retries = retries
+
+    def __enter__(self):
+        sleeptime = 1
+        self.lockpath = "{0}.pid{1}.{2}".format(self.filepath, os.getpid(), LOCKFN)
```

Comment:
Declare `lockpath` here, closer to when it's actually used.  I'm not 100% sure, but I'm guessing if you make a Lock object, and then do a multi-process fork, declaring `lockpath` in the `__init__` will put the parent process ID in the lockpath filename instead of the child process.